### ISSUE TITLE
flip() function for genes containing introns

### ISSUE THREAD

**Issue by xvtyzn** [thackl/gggenomes#91]
Hi again,

I faced the problem of flip function against genes containing intron.
(This is test data, so feel free to use it.)

```R
tmp_gg <- read_gff3("tmp.gff3.txt") %>%
  rename_seq() %>%
  unchop(parent_ids) %>%
  gggenomes() + 
  geom_gene(aes(fill=type, group=parent_ids), size = 5)  +
  scale_fill_brewer(palette = "Set1")

tmp_gg
```

![cd551f25-2564-417c-a067-83cbd0f70345](https://user-images.githubusercontent.com/15158918/137482711-06ae9042-1df7-42a9-9084-aca40252d608.png)

```R
tmp_gg %>% flip_seqs(4)
```

![cc495749-aaa6-4b36-9582-cb7dfd2af960](https://user-images.githubusercontent.com/15158918/137482968-0238d38c-287f-42ba-8e5b-632f36d917c7.png)

The exon does not reverse correctly.

rename_seq function is the following self-made function. It is essentially run to recognize genes that are on the same genome as different bins. It is probably specific to this data and not a common function.

```R
rename_seq <- function(gff_data, gene_list = NA){
  if(!is.na(gene_list)){
    renamed_genes <- gff_data %>%
      dplyr::filter(parent_ids %in% gene_list | 
                      feat_id %in% gene_list) 
  } else {
    renamed_genes <- gff_data %>%
      arrange(seq_id) 
  }
  
  renamed_gff <- renamed_genes %>%
    mutate(seq_ids_psuedo = ifelse(grepl(".m1$",parent_ids), 
                                 str_replace(parent_ids,  
                                             pattern=".m1$", replacement=""), 
[tmp.gff3.txt](https://github.com/thackl/gggenomes/files/7353407/tmp.gff3.txt)

                                 parent_ids)) %>%
    mutate(seq_id = seq_ids_psuedo) %>%
    unchop(seq_id)
  return(renamed_gff)
}
```

I guess it may be related to #64, but I don't know the details.
Thanks in advance.

Keigo

**Comment by thackl** [comment]
Yep, that is definitely off! Right now, there is good and bad news. 

Good news, I'm testing a fix for #65, which seems to work and should make your life easier (will upload soon). 
```
# read_gff3 now can handle Augustus CDS
g0 <- read_gff3("issue_91/tmp.gff3.txt", fix_augustus_cds = TRUE)
# rename_seq() a bit more concise ;)
g1 <- g0 %>% mutate(seq_id = str_remove(map(parent_ids, 1), ".m1$"))

gggenomes(g1) + 
  geom_gene(aes(fill=type), size = 5)  +
  scale_fill_brewer(palette = "Set1")
```
![image](https://user-images.githubusercontent.com/9363087/137538513-684c6ed0-0a0b-42a7-969b-64091ec08cdc.png)

Bad news, fixing `flip()` needs some more thought. The problem are the exons. All other features can simply be flipped by toggling their `strand`. This is also true for CDS because I merge all parts from one CDS into one feature internally. However, exons stay independent features, so instead of flipping all exons of one gene together (like all CDS parts), each exon is flipped on its own, which doesn't give the intended outcome. Need to ponder...

**Comment by biubiu-1** [comment]
Hope the exon flipping issue got fixed someday. That will be really useful to illustrate the introns of eukaryotic genomes.
